### PR TITLE
Remove pods for imagePullBackOff

### PR DESCRIPTION
**Pull request by tnsetting**
# TL;DR
Clean up pod because of imagePullBackOff. This is an alternative for https://github.com/lyft/flytepropeller/pull/212 and of issue: https://github.com/lyft/flyte/issues/651.

We will always clean up the pods failed due to ImagePullBackOff.

The approach is similar to the other PR: When last retry failed in case of ImagePullBackOff the node will still go to EPhaseRetryableFailure and in the handleRetryableFailure function, the pod will be cleaned up and node transit to failed state.

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
 https://github.com/lyft/flyte/issues/651

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Comment by tnsetting** [comment]
@kumare3 Please take a look at the PR as another option for issue: lyft/flyte#651.

**Comment by tnsetting** [comment]
cc @honnix

File: pkg/controller/nodes/executor.go [hunk 1]
```diff
@@ -244,12 +245,15 @@ func (c *nodeExecutor) execute(ctx context.Context, h handler.Node, nCtx *nodeEx
...
 		currentAttempt, maxAttempts, isEligible := c.isEligibleForRetry(nCtx, nodeStatus, phase.GetErr())
 		if !isEligible {
-			return handler.PhaseInfoFailure(
-				core.ExecutionError_USER,
-				fmt.Sprintf("RetriesExhausted|%s", phase.GetErr().Code),
-				fmt.Sprintf("[%d/%d] currentAttempt done. Last Error: %s::%s", currentAttempt, maxAttempts, phase.GetErr().Kind.String(), phase.GetErr().Message),
-				phase.GetInfo(),
-			), nil
+			pErr := phase.GetErr()
+			if pErr.Kind != core.ExecutionError_USER || !strings.Contains(pErr.Code, "ImagePullBackOff") {
```
Comment (tnsetting) [inline pkg/controller/nodes/executor.go:249]: This is not ideal since it breaks the abstraction.

**Comment by tnsetting** [comment]
Close the PR since it is not ideal.
